CarDealerWebAPI/Controllers/VehicleSubmissionsController.cs

```diff
@@ -12,22 +12,26 @@ public class VehicleSubmissionsController : ControllerBase
     {
         private readonly IVehicleSubmissionsService _service;
         private readonly IVehicleService _vehicleService;
+        private readonly IVehicleMarketValueService _marketPrice;
 
-        public VehicleSubmissionsController(IVehicleSubmissionsService service, IVehicleService vehicleService)
+        public VehicleSubmissionsController(IVehicleSubmissionsService service, IVehicleService vehicleService, IVehicleMarketValueService marketPrice)
         {
             _service = service;
             _vehicleService = vehicleService;
+            _marketPrice = marketPrice;
         }
 
         [HttpGet("{UserId}")]
         public IActionResult GetAllVehicleSubmissionsByUser(string UserId)
         {
             return Ok(_service.GetAllVehicleSubmissionsByUser(UserId));
         }
-
+        //this should add to both the vehicle table and the vehicle submission table 
+        //and will need a admin to add it to the vehicle listing table to be seen by everyone.
         [HttpPost]
         public async Task<IActionResult> AddVehicleSubmission(VehicleSubmissions submission)
         {
+            await _marketPrice.GetAverageVehiclePrice(submission.Vehicle.VinNumber);
```

Comment:
Duplicate call to api
In VehicleSubmissionService ln 46:   

`submission.Vehicle.MarketValue =  int.Parse(await _vehicleMarketValueService.GetAverageVehiclePrice(vehicle.VinNumber));`

Reply:
not sure which place is the best for it (service vs controller), but it shouldn't be done twice for each submission